src/Core/Silk.NET.BuildTools/Converters/Readers/OpenCLReader.cs

```diff
@@ -883,65 +942,275 @@ public IEnumerable<Enum> ReadEnums(object obj, BindTask task)
+                        NativeName = name.Original,
+                        Value = value
+                    };
+                }).ToList();
+
+
+
+                string extTag = "";
+                string rawName = "";
+                if (enumExtensions.TryGetValue(group.Key, out var extName))
+                {
+                    extTag = extName.Renamed.Substring(0, extName.Renamed.IndexOf('_'));
+                    var groupNoTag = group.Key.Renamed.Replace($"_{extTag}", "", StringComparison.OrdinalIgnoreCase);
+                    if(!enumEntries.ContainsKey(new RenamedEntry(group.Key.Original, groupNoTag)))
+                    {
+                        rawName = groupNoTag;
+                    }
+                    else
+                    {
+                        rawName = group.Key.Renamed;
+                    }
+                }
+                else
+                {
+                    extName = new RenamedEntry("Core (Grouped)", "Core (Grouped)");
+                    rawName = group.Key.Renamed;
+                }
+
+                var groupName = Naming.Translate(TrimName(rawName, task), task.FunctionPrefix);
+                var is64bit = ulongEnums.Contains(group.Key.Renamed);
+
+                RenameTokens(tokens, group.Key.Renamed, extTag, task);
+
+                foreach (var (apiName, apiVersion) in registry
+                    .Elements("feature")
+                    .Select(x => (x.Attribute("api")?.Value, x.Attribute("number")?.Value))
+                    .Distinct())
+                {
+                    var ret = new Enum
+                    {
+                        Name = groupName,
+                        NativeName = group.Key.Original,
+                        Attributes = GetEnumAttributes(group.Key.Renamed),
+                        ExtensionName = extName.Renamed,
+                        ProfileName = apiName,
+                        ProfileVersion = Version.Parse(apiVersion),
+                        Tokens = tokens,
+                        EnumBaseType = is64bit ? new Type { Name = "ulong" } : new Type { Name = "int" }
+                    };
+            
+                    yield return ret;
+                }
+            }
+        }
+
+        private string FindCommonPrefix(List<string> names, bool containsTypeName)
+        {
+            var commonPrefixFirstPass = FindCommonPrefix(names, containsTypeName, names.Max(x => x.Length));
+            var tgtPos = commonPrefixFirstPass.Length;
+
+            var startingWithDigit = names.Where(n => n.Length > tgtPos && char.IsDigit(n[tgtPos]));
+            if (startingWithDigit.Any())
+            {
+                return FindCommonPrefix(names, containsTypeName, tgtPos - 1);
+            }
+
+            return commonPrefixFirstPass;
+        }
+
+        private string FindCommonPrefix(List<string> names, bool containsTypeName, int maxPos)
```

Comment:
These two methods are genius! Can you move them into a common place (such as `Utilities`) so we can use them elsewhere in the generator? Preferably documented, too.